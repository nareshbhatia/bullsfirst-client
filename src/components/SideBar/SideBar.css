.sidebar {
  flex: 0 0 200px;
  display: flex;
  flex-direction: column;
  background-color: var(--bg-sidebar);
}

.sidebar__title {
  font-weight: 600;
  font-size: 18px;
  line-height: 22px;
  color: var(--text-primary-default);
  margin: var(--spacing-1) 0 0 0;
}

.sidebar__items {
  list-style: none;
  padding-left: 0;
  margin: var(--spacing-2) 0;
}

.sidebar__item {
  font-size: 14px;
  line-height: 18px;
  color: var(--sidebar-item);
  margin-bottom: var(--spacing-2);
  cursor: pointer;
}

.sidebar__item--active {
  color: var(--sidebar-item-ative);
}
