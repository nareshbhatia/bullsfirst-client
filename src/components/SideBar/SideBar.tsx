import React from 'react';
import { ScrollingContainer } from '../Containers';
import './SideBar.css';

export interface NavItem {
  id: string;
  name: string;
}

export interface SideBarProps {
  title?: string;
  items?: Array<NavItem>;
  selectedNavItemId: string;
  onNavItemSelected: (navItemId: string) => void;
}

export const SideBar = ({
  title,
  items,
  selectedNavItemId,
  onNavItemSelected,
}: SideBarProps) => {
  return (
    <div data-testid="sidebar" className="sidebar">
      <ScrollingContainer className="flex-1 p-2">
        {title !== undefined ? (
          <h1 className="sidebar__title">{title}</h1>
        ) : null}
        {items !== undefined ? (
          <ul className="sidebar__items">
            {items.map((item) => (
              <li
                key={item.id}
                className={`sidebar__item
                ${item.id === selectedNavItemId ? 'sidebar__item--active' : ''}
              `}
                onClick={() => onNavItemSelected(item.id)}
              >
                {item.name}
              </li>
            ))}
          </ul>
        ) : null}
      </ScrollingContainer>
    </div>
  );
};
