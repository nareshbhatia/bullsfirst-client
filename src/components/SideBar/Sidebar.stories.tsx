import React, { useState } from 'react';
import { Story, Meta } from '@storybook/react';
import {
  HorizontalContainer,
  ScrollingContainer,
  ViewVerticalContainer,
} from '../Containers';
import { SimpleHeader } from '../Header';
import { NavItem, SideBar } from './SideBar';

const items: Array<NavItem> = [
  {
    id: 'brokerage-account',
    name: 'Brokerage Account',
  },
  {
    id: 'retirement-account',
    name: 'Retirement Account',
  },
  {
    id: 'jennys-college-fund',
    name: "Jenny's College Fund",
  },
];

export default {
  title: 'Components/SideBar',
  component: SideBar,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta;

const Template: Story = () => {
  const [selectedNavId, setSelectedNavId] = useState('brokerage-account');

  const handleNavItemSelected = (navItemId: string) => {
    setSelectedNavId(navItemId);
  };

  return (
    <ViewVerticalContainer>
      <SimpleHeader />
      <HorizontalContainer className="min-h-0">
        <SideBar
          title="Accounts"
          items={items}
          selectedNavItemId={selectedNavId}
          onNavItemSelected={handleNavItemSelected}
        />
        <ScrollingContainer className="flex-1 p-2">
          <h1 className="title2">
            {items.find((item) => item.id === selectedNavId)?.name}
          </h1>
        </ScrollingContainer>
      </HorizontalContainer>
    </ViewVerticalContainer>
  );
};

export const SideBarStory = Template.bind({});
SideBarStory.storyName = 'SideBar';
SideBarStory.args = {};
