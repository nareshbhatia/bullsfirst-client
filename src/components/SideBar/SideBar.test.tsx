import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { NavItem, SideBar } from './SideBar';

const handleNavItemSelected = jest.fn();

const items: Array<NavItem> = [
  {
    id: 'brokerage-account',
    name: 'Brokerage Account',
  },
  {
    id: 'retirement-account',
    name: 'Retirement Account',
  },
  {
    id: 'jennys-college-fund',
    name: "Jenny's College Fund",
  },
];

describe('SideBar', () => {
  beforeEach(() => {
    handleNavItemSelected.mockReset();
  });

  it('renders correctly without a title', () => {
    const { asFragment } = render(
      <SideBar
        items={items}
        selectedNavItemId="brokerage-account"
        onNavItemSelected={handleNavItemSelected}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly without items', () => {
    const { asFragment } = render(
      <SideBar
        title="Accounts"
        selectedNavItemId="brokerage-account"
        onNavItemSelected={handleNavItemSelected}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onNavItemSelected when a nav item is clicked', () => {
    const { getByText } = render(
      <SideBar
        title="Accounts"
        items={items}
        selectedNavItemId="brokerage-account"
        onNavItemSelected={handleNavItemSelected}
      />
    );

    fireEvent.click(getByText('Retirement Account'));
    expect(handleNavItemSelected).toBeCalledWith('retirement-account');
  });
});
