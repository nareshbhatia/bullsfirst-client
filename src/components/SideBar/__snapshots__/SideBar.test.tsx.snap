// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders correctly without a title 1`] = `
<DocumentFragment>
  <div
    class="sidebar"
    data-testid="sidebar"
  >
    <div
      class="overflow-auto flex-1 p-2"
      data-testid="scrolling-container"
    >
      <ul
        class="sidebar__items"
      >
        <li
          class="sidebar__item
                sidebar__item--active
              "
        >
          Brokerage Account
        </li>
        <li
          class="sidebar__item
                
              "
        >
          Retirement Account
        </li>
        <li
          class="sidebar__item
                
              "
        >
          Jenny's College Fund
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SideBar renders correctly without items 1`] = `
<DocumentFragment>
  <div
    class="sidebar"
    data-testid="sidebar"
  >
    <div
      class="overflow-auto flex-1 p-2"
      data-testid="scrolling-container"
    >
      <h1
        class="sidebar__title"
      >
        Accounts
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
